hockeysdk/src/main/java/net/hockeyapp/android/CrashManager.java

```diff
@@ -403,64 +403,72 @@ public static synchronized void submitStackTraces(final WeakReference<Context> w
     }
 
     private static void submitStackTrace(final WeakReference<Context> weakContext, String filename, CrashManagerListener listener, CrashMetaData crashMetaData) {
+        String stacktrace = null;
+        try {
+            stacktrace = contentsOfFile(weakContext, filename);
+        } catch (Exception e) {
+            HockeyLog.error("Failed to read crash data", e);
+        }
+        if (stacktrace.length() == 0) {
+            HockeyLog.warn("The crash data is invalid");
+            deleteStackTrace(weakContext, filename);
+            if (listener != null) {
+                listener.onCrashesNotSent();
```

Comment:
On more thing I don't understand is that we can have multiple stack traces.
This method is called in a loop, yet at first invalid stack trace you are calling listener. Isn't that the wrong thing to do?

Reply:
Edit: actually we call it for each crashes on success but no parameter is passed, I guess design issue and I was confused as in AppCenter we pass the report as a parameter for those.